test/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/CheckAllAnswersAndSubmitControllerSpec.scala

```diff
@@ -485,22 +491,132 @@ class CheckAllAnswersAndSubmitControllerSpec
+            AmountInPence(0),
+            "",
+            noTaxDueRefLine,
+            List(submissionLine, addressLine)
+          ),
+          TestScenario(
+            "user with tax due",
+            Some(UserType.Individual),
+            AmountInPence(10000),
+            "",
+            taxDueRefLine,
+            List(submissionLine, returnReferenceWithBundleId, addressLine, taxDueDateLine)
+          ),
+          TestScenario(
+            "agent with no tax due",
+            Some(UserType.Agent),
+            AmountInPence(0),
+            "Client: ",
+            noTaxDueRefLine,
+            List(submissionLine, addressLine)
+          ),
+          TestScenario(
+            "agent with tax due",
+            Some(UserType.Agent),
+            AmountInPence(10000),
+            "Client: ",
+            taxDueRefLine,
+            List(submissionLine, returnReferenceWithBundleId, addressLine, taxDueDateLine)
+          ),
+          TestScenario(
+            "organisation with no tax due",
+            Some(UserType.Organisation),
+            AmountInPence(0),
+            "Trust: ",
+            noTaxDueRefLine,
+            List(submissionLine, addressLine)
+          ),
+          TestScenario(
+            "organisation with tax due",
+            Some(UserType.Organisation),
+            AmountInPence(10000),
+            "Trust: ",
+            taxDueRefLine,
+            List(submissionLine, returnReferenceWithBundleId, addressLine, taxDueDateLine)
+          )
+        )
+
+        scenarios.foreach {
+          case TestScenario(description, userType, taxOwed, namePrefix, reference, expectedTable) => {
+            withClue(description) {
+              val address = sample[UkAddress].copy(
+                line1    = "123 fake street",
+                line2    = None,
+                town     = None,
+                county   = None,
+                postcode = Postcode("abc123")
+              )
+              val processingDate = LocalDate.of(2020, 2, 2)
+              val dueDate        = LocalDate.of(2020, 1, 1)
+              val returnResponse = sample[SubmitReturnResponse].copy(
+                formBundleId   = "form bundle id",
+                processingDate = LocalDateTime.of(processingDate, LocalTime.of(1, 1)),
+                charge = Some(
+                  sample[ReturnCharge].copy(
+                    amount          = taxOwed,
+                    chargeReference = "charge ref",
+                    dueDate         = dueDate
+                  )
+                )
+              )
+              val subscribe = sample[SubscribedDetails].copy(name = Right(IndividualName("John", "Doe")))
+              val justSubmittedReturn = sample[JustSubmittedReturn].copy(
+                submissionResponse = returnResponse,
+                subscribedDetails  = subscribe,
+                completeReturn     = sample[CompleteSingleDisposalReturn].copy(propertyAddress = address)
```

Comment:
if we have a user type of agent can we also populate the agentReference number with some value here for consistency? So something like 
```scala
 .copy(
    ...
    agentReferenceNumber = if(userType === UserType.Agent) Some(sample[AgentReferenceNumber]) else None
)
```